umbrella/src/umbrella.py

```diff
@@ -380,41 +380,47 @@ def dependency_process(env_para_dict, name, id, mountpoint, action, packages_jso
 	mount_value = ''
 	local_cvmfs = ''
 
-	if name[:5] == 'cmssw':
-		is_cms_app = 1
-
-		local_cms = check_cvmfs_cms()
-		if local_cms:
-			local_cvmfs = os.path.dirname(local_cms)
-			logging.debug("The cvmfs is installed on the local host, and its mountpoint is: %s", local_cvmfs)
-		else:
-			logging.debug("The cvmfs is not installed on the local host.")
+	item = package_search(packages_json, name, id)
+	#table schema: (name text, version text, platform text, store text, store_type text, type text)
+	store = item["source"][0]
+	logging.debug("%s is chosen to deliver %s", store, name)
 
-		#if local_cvmfs is empty, set $HTTP_PROXY, download cctools package, set cvmfs_siteconf_mountpoint, and call parrotize_user_cmd.
-		if not local_cvmfs:
-			logging.debug("Add env variable HTTP_PROXY = http://cache01.hep.wisc.edu:3128 into env_para_dict")
-			env_para_dict["HTTP_PROXY"] = "http://cache01.hep.wisc.edu:3128"
+	if name[:5] == 'cmssw': # another solution: split name based on '-' and check whether the first item is 'cmssw'.
+		is_cms_app = 1
```

Comment:
There exist software systems other than CMS that are delivered by cvmfs: atlas, alice, minos, etc.
Will this work for any software, or just for things named "cmssw"?